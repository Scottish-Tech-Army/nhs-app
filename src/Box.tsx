import React, { FormEvent, useState } from "react";
import { TRAUMA_TOWER_TEMPLATE } from "./data/traumaTower";
import { useParams } from "react-router-dom";

export type FormValueType = {
  name: string;
  count: number;
};

export interface BoxProps {
  setBoxContents: (boxTemplateId: string, boxNumber: number, contents: FormValueType[]) => void;
}

function Box({ setBoxContents }: BoxProps) {

//init variables
  //destructure nested routes from useParams  
  let { boxTemplateId, boxId } = useParams();
  //init boxTemplate var finding boxTemplateId from TRAUMA_TOWER_TEMPLATE Doc
  const boxTemplate = TRAUMA_TOWER_TEMPLATE.boxes.find(
    (box) => box.boxTemplateId === boxTemplateId
  );
  //init var of mapped items in template creating obj of name and count/amount of each
  const startingArrayOfObjects: FormValueType[] =
      boxTemplate?.items.map(({ name }) => ({ name, count: 0 })) || [];
  //init- useState hook using mapped items stored as init value  
  const [formValues, setFormValues] = useState<FormValueType[]>(
    startingArrayOfObjects
  );
  //init boxNumber to zero   
    let boxNumber = 0;
  //check boxId matches string of starting digit/end digit  
    //parse boxId as Int
    //it should so continue(maybe integrate fail condition?)
  if (boxId?.match(/^\d+$/)) {
    boxNumber = Number.parseInt(boxId);
  }
  //check- no boxTemplate OR boxNumber less than 1 OR boxNumber is more than boxTemplate count/value
    //if true return null (probably should return Error)
    //if false continue
  if (!boxTemplate || boxNumber < 1 || boxNumber > boxTemplate.count) {
    return null;
  }

  //Functions for storing form values on user input- form change/submit
    //handle change- capture change to useState, set change to setFormValues
        //init- form new form values to array
        //process values to Int
        //set to form values state
  let handleChange = (index: number, target: HTMLInputElement) => {
    let newFormValues = [...formValues];

    newFormValues[index].count = Number.parseInt(target!.value);

    setFormValues(newFormValues);
  };

   //handleSubmit- pass router values and inputted state values  
        //handle browser event to keep inputted data
        //pass router values and inputted form values to setBoxContents function
  let handleSubmit = (event: FormEvent<HTMLFormElement>) => {
    event.preventDefault();
    setBoxContents(boxTemplateId!, boxNumber, formValues);
  };

  return (
    <div>
      <h1>{`${boxTemplate.name} - Box ${boxId}`}</h1>

      <form onSubmit={handleSubmit}>
        <ul>
          {boxTemplate.items.map((item, index) => (
            <li key={index}>
              <label>
                {item.name}
                <input
                  name={item.name}
                  type="number"
                  max="10"
                  min="0"
                  step="1"
                  value={formValues[index].count || 0}
                  onChange={({ target }) => handleChange(index, target)}
                />
              </label>
            </li>
          ))}
        </ul>
        <div>
          <button type="submit">Submit</button>
        </div>
      </form>
    </div>
  );
}

export default Box;
