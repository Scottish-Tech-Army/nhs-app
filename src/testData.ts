import { BoxContents, StorageAreaContents } from "./model/StorageTypes";
import { TRAUMA_TOWER_TEMPLATE } from "./model/TraumaTower";

export const FULL_CHEST_DRAIN_BOX: Omit<BoxContents, "boxNumber"> = {
  boxTemplateId: "0",
  items: [
    {
      name: "Blunt dissection chest drainage insertion pack",
      size: "28Fg",
      quantity: 1,
    },
    { name: "Sterile gloves", size: "Small", quantity: 1 },
    { name: "Sterile gloves", size: "Medium", quantity: 1 },
    { name: "Sterile gloves", size: "Large", quantity: 1 },
    { name: "Chest drain catheter", size: "28Fr", quantity: 1 },
    { name: "Chest drain catheter", size: "32Fr", quantity: 1 },
    { name: "Chest drain catheter", size: "36Fr", quantity: 1 },
    { name: "ChloraPrep applicator", quantity: 2 },
    { name: "Lidocaine 1%", size: "5ml / 50mg", quantity: 2 },
    { name: "Standard suture pack", size: "Standard", quantity: 1 },
    { name: "Mefix roll", size: "5cm x 10m", quantity: 1 },
    { name: "Chest drain bottle", quantity: 1 },
    { name: "Chest drain tubing", quantity: 1 },
    {
      name: "Sterile water (H20) bottle",
      size: "1000ml bottle",
      quantity: 1,
    },
    { name: "Spencer wells forceps", size: "Straight 20cm", quantity: 1 },
  ],
};

export const ZERO_CHEST_DRAIN_BOX: Omit<BoxContents, "boxNumber"> = {
  boxTemplateId: "0",
  items: [
    {
      name: "Blunt dissection chest drainage insertion pack",
      size: "28Fg",
      quantity: 0,
    },
    { name: "Sterile gloves", size: "Small", quantity: 0 },
    { name: "Sterile gloves", size: "Medium", quantity: 0 },
    { name: "Sterile gloves", size: "Large", quantity: 0 },
    { name: "Chest drain catheter", size: "28Fr", quantity: 0 },
    { name: "Chest drain catheter", size: "32Fr", quantity: 0 },
    { name: "Chest drain catheter", size: "36Fr", quantity: 0 },
    { name: "ChloraPrep applicator", quantity: 0 },
    { name: "Lidocaine 1%", size: "5ml / 50mg", quantity: 0 },
    { name: "Standard suture pack", size: "Standard", quantity: 0 },
    { name: "Mefix roll", size: "5cm x 10m", quantity: 0 },
    { name: "Chest drain bottle", quantity: 0 },
    { name: "Chest drain tubing", quantity: 0 },
    {
      name: "Sterile water (H20) bottle",
      size: "1000ml bottle",
      quantity: 0,
    },
    { name: "Spencer wells forceps", size: "Straight 20cm", quantity: 0 },
  ],
};

export const EMPTY_CHEST_DRAIN_BOX: Omit<BoxContents, "boxNumber"> = {
  boxTemplateId: "0",
  items: [],
};

export const ALL_CONTENTS: StorageAreaContents = {
  storageAreaId: TRAUMA_TOWER_TEMPLATE.storageAreaId,
  boxes: [
    { ...FULL_CHEST_DRAIN_BOX, boxNumber: 1 },
    { ...FULL_CHEST_DRAIN_BOX, boxNumber: 2 },
    { ...FULL_CHEST_DRAIN_BOX, boxNumber: 3 },
    { ...FULL_CHEST_DRAIN_BOX, boxNumber: 4 },
    { ...FULL_CHEST_DRAIN_BOX, boxNumber: 5 },
    { ...FULL_CHEST_DRAIN_BOX, boxNumber: 6 },
  ],
};

export const PARTIAL_CONTENTS: StorageAreaContents = {
  storageAreaId: TRAUMA_TOWER_TEMPLATE.storageAreaId,
  boxes: [
    { ...FULL_CHEST_DRAIN_BOX, boxNumber: 1 },
    { ...FULL_CHEST_DRAIN_BOX, boxNumber: 2 },

    {
      boxTemplateId: "0",
      boxNumber: 3,
      items: [
        {
          name: "Blunt dissection chest drainage insertion pack",
          size: "28Fg",
          quantity: 1,
        },
        { name: "Sterile gloves", size: "Small", quantity: 1 },
        { name: "Sterile gloves", size: "Medium", quantity: 0 },
        { name: "Sterile gloves", size: "Large", quantity: 0 },
        { name: "Chest drain catheter", size: "36Fr", quantity: 1 },
        { name: "Lidocaine 1%", size: "5ml / 50mg", quantity: 1 }, // 1 missing
        { name: "Standard suture pack", size: "Standard", quantity: 3 }, // 2 extra
        // TODO - check having too many in the box is a bad thing
        { name: "Mefix roll", size: "5cm x 10m", quantity: 1 },
        { name: "Spencer wells forceps", size: "Straight 20cm", quantity: 1 },
      ],
    },
    { ...ZERO_CHEST_DRAIN_BOX, boxNumber: 4 },
    { ...ZERO_CHEST_DRAIN_BOX, boxNumber: 5 },
    { ...ZERO_CHEST_DRAIN_BOX, boxNumber: 6 },
  ],
};

export const EMPTY_CONTENTS: StorageAreaContents = {
  storageAreaId: TRAUMA_TOWER_TEMPLATE.storageAreaId,
  boxes: [
    { ...EMPTY_CHEST_DRAIN_BOX, boxNumber: 1 },
    { ...EMPTY_CHEST_DRAIN_BOX, boxNumber: 2 },
    { ...EMPTY_CHEST_DRAIN_BOX, boxNumber: 3 },
    { ...EMPTY_CHEST_DRAIN_BOX, boxNumber: 4 },
    { ...EMPTY_CHEST_DRAIN_BOX, boxNumber: 5 },
    { ...EMPTY_CHEST_DRAIN_BOX, boxNumber: 6 },
  ],
};

export const ZERO_CONTENTS: StorageAreaContents = {
  storageAreaId: TRAUMA_TOWER_TEMPLATE.storageAreaId,
  boxes: [
    { ...ZERO_CHEST_DRAIN_BOX, boxNumber: 1 },
    { ...ZERO_CHEST_DRAIN_BOX, boxNumber: 2 },
    { ...ZERO_CHEST_DRAIN_BOX, boxNumber: 3 },
    { ...ZERO_CHEST_DRAIN_BOX, boxNumber: 4 },
    { ...ZERO_CHEST_DRAIN_BOX, boxNumber: 5 },
    { ...ZERO_CHEST_DRAIN_BOX, boxNumber: 6 },
  ],
};
