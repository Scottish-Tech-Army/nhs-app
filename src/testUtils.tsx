import React from "react";
import { render } from "@testing-library/react";
import type { RenderOptions } from "@testing-library/react";
import { configureStore } from "@reduxjs/toolkit";
import type { PreloadedState } from "@reduxjs/toolkit";
import { Provider } from "react-redux";

import type { AppStore, RootState } from "./data/store";
import boxContentsReducer, { initialState } from "./data/BoxContentsSlice";

interface ExtendedRenderOptions extends Omit<RenderOptions, "queries"> {
  preloadedState?: PreloadedState<RootState>;
  store?: AppStore;
}

export function renderWithProvider(
  component: React.ReactElement,
  {
    preloadedState = { boxContents: initialState },
    store = configureStore({
      reducer: { boxContents: boxContentsReducer },
      preloadedState,
    }),
    ...renderOptions
  }: ExtendedRenderOptions = {}
) {

  return {
    store,
    ...render(<Provider store={store}>{component}</Provider>, renderOptions),
  };
}
