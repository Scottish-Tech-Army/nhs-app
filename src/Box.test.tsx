/* eslint-disable testing-library/no-node-access */
/* eslint-disable testing-library/prefer-screen-queries */
import React from "react";

import { getByRole, screen } from "@testing-library/react";
import { Routes, Route } from "react-router-dom";

import Box from "./Box";
import { FULL_CHEST_DRAIN_BOX } from "./testData";
import { renderWithProvider } from "./testUtils";


describe("Box", () => {
  it("rendered a box page", async () => {
    renderWithRoute("0", "3");
    expect(screen.getByText("Trauma Chest Drain - Box 3")).toBeDefined();

    const inputFields = Array.from(document.querySelectorAll(".display-name"));

    expect(inputFields.map((input) => input.textContent)).toEqual([
      "Blunt dissection chest drainage insertion pack (28Fg)",
      "Sterile gloves (Small)",
      "Sterile gloves (Medium)",
      "Sterile gloves (Large)",
      "Chest drain catheter (28Fr)",
      "Chest drain catheter (32Fr)",
      "Chest drain catheter (36Fr)",
      "ChloraPrep applicator",
      "Lidocaine 1% (5ml / 50mg)",
      "Standard suture pack (Standard)",
      "Mefix roll (5cm x 10m)",
      "Chest drain bottle",
      "Chest drain tubing",
      "Sterile water (H20) bottle (1000ml bottle)",
      "Spencer wells forceps (Straight 20cm)",
    ]);

    expect(screen.getByRole("button", { name: "Save" })).toBeDefined();
  });

  it("does not render if no boxTemplateId", async () => {
    const { container } = renderWithRoute("", "3");

    expect(container).toHaveTextContent("Unknown path");
  });

  it("does not render if no boxId", async () => {
    const { container } = renderWithRoute("0", "");

    expect(container).toHaveTextContent("Unknown path");
  });

  it("does not render if unknown boxTemplateId", async () => {
    const { container } = renderWithRoute("Unknown", "4");

    expect(container.children).toHaveLength(0);

    expect(container).toMatchSnapshot();
  });

  it("does not render if unknown boxId", async () => {
    const { container } = renderWithRoute("0", "Unknown");

    expect(container.children).toHaveLength(0);

    expect(container).toMatchSnapshot();
  });

  it("does not render if unknown boxId - box number too high", async () => {
    const { container } = renderWithRoute("0", "7");

    expect(container.children).toHaveLength(0);

    expect(container).toMatchSnapshot();
  });

  it("can save item changes", async () => {
    const { store, user, history } = renderWithRoute("0", "3");

    const itemLabel = screen.getByText(
      "Blunt dissection chest drainage insertion pack (28Fg)"
    );

    const increaseButton = getByRole(itemLabel.parentElement!, "button", {
      name: "add item",
    });
    const decreaseButton = getByRole(itemLabel.parentElement!, "button", {
      name: "remove item",
    });
    await user.click(increaseButton);
    await user.click(increaseButton);
    await user.click(increaseButton);
    await user.click(decreaseButton);

    await user.click(screen.getByRole("button", { name: "Save" }));

    expect(store.getState().boxContents.boxes[2].items).toEqual([
      {
        quantity: 2,
        name: "Blunt dissection chest drainage insertion pack",
        size: "28Fg",
      },
      { quantity: 0, name: "Sterile gloves", size: "Small" },
      { quantity: 0, name: "Sterile gloves", size: "Medium" },
      { quantity: 0, name: "Sterile gloves", size: "Large" },
      { quantity: 0, name: "Chest drain catheter", size: "28Fr" },
      { quantity: 0, name: "Chest drain catheter", size: "32Fr" },
      { quantity: 0, name: "Chest drain catheter", size: "36Fr" },
      { quantity: 0, name: "ChloraPrep applicator" },
      { quantity: 0, name: "Lidocaine 1%", size: "5ml / 50mg" },
      { quantity: 0, name: "Standard suture pack", size: "Standard" },
      { quantity: 0, name: "Mefix roll", size: "5cm x 10m" },
      { quantity: 0, name: "Chest drain bottle" },
      { quantity: 0, name: "Chest drain tubing" },
      {
        quantity: 0,
        name: "Sterile water (H20) bottle",
        size: "1000ml bottle",
      },
      { quantity: 0, name: "Spencer wells forceps", size: "Straight 20cm" },
    ]);

    expect(history.location.pathname).toEqual("/");
  });

  it("can mark box as full", async () => {
    const { store, user, history } = renderWithRoute("0", "3");

    await user.click(screen.getByRole("button", { name: "FULL" }));

    expect(store.getState().boxContents.boxes[2]).toEqual({
      ...FULL_CHEST_DRAIN_BOX,
      boxNumber: 3,
    });

    expect(history.location.pathname).toEqual("/");
  });

  it("can go to item details", async () => {
    const { user, history } = renderWithRoute("0", "3");

    const itemLabel = screen.getByText("Chest drain catheter (28Fr)");

    const infoButton = getByRole(itemLabel.parentElement!, "button", {
      name: "item information",
    });
    await user.click(infoButton);

    expect(history.location.pathname).toEqual("/item/0/4");
  });

  it("can return to previous page", async () => {
    const { user, history } = renderWithRoute("0", "3");

    await user.click(screen.getByRole("button", { name: "Back" }));

    expect(history.location.pathname).toEqual("/");
  });

  it("renders correctly", () => {
    const { container } = renderWithRoute("0", "3");

    expect(container).toMatchSnapshot();
  });

  function renderWithRoute(boxTemplateId: string, boxId: string) {
    // Add routes to get the contents of useParams populated
    return renderWithProvider(
      <Routes>
        <Route path="box/:boxTemplateId/:boxId" element={<Box />} />
        <Route path="*" element={<div>Unknown path</div>} />
      </Routes>,
      { initialRoutes: ["/", `/box/${boxTemplateId}/${boxId}`] }
    );
  }
});
