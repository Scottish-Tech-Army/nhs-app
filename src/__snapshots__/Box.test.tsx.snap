// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box does not render if unknown boxId - box number too high 1`] = `<div />`;

exports[`Box does not render if unknown boxId 1`] = `<div />`;

exports[`Box does not render if unknown boxTemplateId 1`] = `<div />`;

exports[`Box renders correctly 1`] = `
<div>
  <div
    class="box-details"
  >
    <header>
      <h1>
        Trauma Chest Drain - Box 3
      </h1>
      <button
        aria-label="Back"
        class="back"
        type="button"
      >
        <svg>
          arrow-left.svg
        </svg>
      </button>
      <button
        class="full"
        type="button"
      >
        FULL
      </button>
    </header>
    <main>
      <div
        class="scroll"
      >
        <div
          class="item"
        >
          <div
            class="display-name"
          >
            Blunt dissection chest drainage insertion pack (28Fg)
          </div>
          <div
            class="controls"
          >
            <button
              aria-label="item information"
              class="item-info"
              type="button"
            >
              <svg>
                info-circle.svg
              </svg>
            </button>
            <div
              class="quantity"
            >
              <span
                class="actual-quantity-bad"
              >
                0
              </span>
              <span>
                 / 
              </span>
              <span
                class="expected-quantity"
              >
                1
              </span>
            </div>
            <button
              aria-label="remove item"
              class="change-quantity"
              disabled=""
              type="button"
            >
              <svg>
                minus.svg
              </svg>
            </button>
            <button
              aria-label="add item"
              class="change-quantity"
              type="button"
            >
              <svg>
                add.svg
              </svg>
            </button>
          </div>
        </div>
        <div
          class="item"
        >
          <div
            class="display-name"
          >
            Sterile gloves (Small)
          </div>
          <div
            class="controls"
          >
            <button
              aria-label="item information"
              class="item-info"
              type="button"
            >
              <svg>
                info-circle.svg
              </svg>
            </button>
            <div
              class="quantity"
            >
              <span
                class="actual-quantity-bad"
              >
                0
              </span>
              <span>
                 / 
              </span>
              <span
                class="expected-quantity"
              >
                1
              </span>
            </div>
            <button
              aria-label="remove item"
              class="change-quantity"
              disabled=""
              type="button"
            >
              <svg>
                minus.svg
              </svg>
            </button>
            <button
              aria-label="add item"
              class="change-quantity"
              type="button"
            >
              <svg>
                add.svg
              </svg>
            </button>
          </div>
        </div>
        <div
          class="item"
        >
          <div
            class="display-name"
          >
            Sterile gloves (Medium)
          </div>
          <div
            class="controls"
          >
            <button
              aria-label="item information"
              class="item-info"
              type="button"
            >
              <svg>
                info-circle.svg
              </svg>
            </button>
            <div
              class="quantity"
            >
              <span
                class="actual-quantity-bad"
              >
                0
              </span>
              <span>
                 / 
              </span>
              <span
                class="expected-quantity"
              >
                1
              </span>
            </div>
            <button
              aria-label="remove item"
              class="change-quantity"
              disabled=""
              type="button"
            >
              <svg>
                minus.svg
              </svg>
            </button>
            <button
              aria-label="add item"
              class="change-quantity"
              type="button"
            >
              <svg>
                add.svg
              </svg>
            </button>
          </div>
        </div>
        <div
          class="item"
        >
          <div
            class="display-name"
          >
            Sterile gloves (Large)
          </div>
          <div
            class="controls"
          >
            <button
              aria-label="item information"
              class="item-info"
              type="button"
            >
              <svg>
                info-circle.svg
              </svg>
            </button>
            <div
              class="quantity"
            >
              <span
                class="actual-quantity-bad"
              >
                0
              </span>
              <span>
                 / 
              </span>
              <span
                class="expected-quantity"
              >
                1
              </span>
            </div>
            <button
              aria-label="remove item"
              class="change-quantity"
              disabled=""
              type="button"
            >
              <svg>
                minus.svg
              </svg>
            </button>
            <button
              aria-label="add item"
              class="change-quantity"
              type="button"
            >
              <svg>
                add.svg
              </svg>
            </button>
          </div>
        </div>
        <div
          class="item"
        >
          <div
            class="display-name"
          >
            Chest drain catheter (28Fr)
          </div>
          <div
            class="controls"
          >
            <button
              aria-label="item information"
              class="item-info"
              type="button"
            >
              <svg>
                info-circle.svg
              </svg>
            </button>
            <div
              class="quantity"
            >
              <span
                class="actual-quantity-bad"
              >
                0
              </span>
              <span>
                 / 
              </span>
              <span
                class="expected-quantity"
              >
                1
              </span>
            </div>
            <button
              aria-label="remove item"
              class="change-quantity"
              disabled=""
              type="button"
            >
              <svg>
                minus.svg
              </svg>
            </button>
            <button
              aria-label="add item"
              class="change-quantity"
              type="button"
            >
              <svg>
                add.svg
              </svg>
            </button>
          </div>
        </div>
        <div
          class="item"
        >
          <div
            class="display-name"
          >
            Chest drain catheter (32Fr)
          </div>
          <div
            class="controls"
          >
            <button
              aria-label="item information"
              class="item-info"
              type="button"
            >
              <svg>
                info-circle.svg
              </svg>
            </button>
            <div
              class="quantity"
            >
              <span
                class="actual-quantity-bad"
              >
                0
              </span>
              <span>
                 / 
              </span>
              <span
                class="expected-quantity"
              >
                1
              </span>
            </div>
            <button
              aria-label="remove item"
              class="change-quantity"
              disabled=""
              type="button"
            >
              <svg>
                minus.svg
              </svg>
            </button>
            <button
              aria-label="add item"
              class="change-quantity"
              type="button"
            >
              <svg>
                add.svg
              </svg>
            </button>
          </div>
        </div>
        <div
          class="item"
        >
          <div
            class="display-name"
          >
            Chest drain catheter (36Fr)
          </div>
          <div
            class="controls"
          >
            <button
              aria-label="item information"
              class="item-info"
              type="button"
            >
              <svg>
                info-circle.svg
              </svg>
            </button>
            <div
              class="quantity"
            >
              <span
                class="actual-quantity-bad"
              >
                0
              </span>
              <span>
                 / 
              </span>
              <span
                class="expected-quantity"
              >
                1
              </span>
            </div>
            <button
              aria-label="remove item"
              class="change-quantity"
              disabled=""
              type="button"
            >
              <svg>
                minus.svg
              </svg>
            </button>
            <button
              aria-label="add item"
              class="change-quantity"
              type="button"
            >
              <svg>
                add.svg
              </svg>
            </button>
          </div>
        </div>
        <div
          class="item"
        >
          <div
            class="display-name"
          >
            ChloraPrep applicator
          </div>
          <div
            class="controls"
          >
            <button
              aria-label="item information"
              class="item-info"
              type="button"
            >
              <svg>
                info-circle.svg
              </svg>
            </button>
            <div
              class="quantity"
            >
              <span
                class="actual-quantity-bad"
              >
                0
              </span>
              <span>
                 / 
              </span>
              <span
                class="expected-quantity"
              >
                2
              </span>
            </div>
            <button
              aria-label="remove item"
              class="change-quantity"
              disabled=""
              type="button"
            >
              <svg>
                minus.svg
              </svg>
            </button>
            <button
              aria-label="add item"
              class="change-quantity"
              type="button"
            >
              <svg>
                add.svg
              </svg>
            </button>
          </div>
        </div>
        <div
          class="item"
        >
          <div
            class="display-name"
          >
            Lidocaine 1% (5ml / 50mg)
          </div>
          <div
            class="controls"
          >
            <button
              aria-label="item information"
              class="item-info"
              type="button"
            >
              <svg>
                info-circle.svg
              </svg>
            </button>
            <div
              class="quantity"
            >
              <span
                class="actual-quantity-bad"
              >
                0
              </span>
              <span>
                 / 
              </span>
              <span
                class="expected-quantity"
              >
                2
              </span>
            </div>
            <button
              aria-label="remove item"
              class="change-quantity"
              disabled=""
              type="button"
            >
              <svg>
                minus.svg
              </svg>
            </button>
            <button
              aria-label="add item"
              class="change-quantity"
              type="button"
            >
              <svg>
                add.svg
              </svg>
            </button>
          </div>
        </div>
        <div
          class="item"
        >
          <div
            class="display-name"
          >
            Standard suture pack (Standard)
          </div>
          <div
            class="controls"
          >
            <button
              aria-label="item information"
              class="item-info"
              type="button"
            >
              <svg>
                info-circle.svg
              </svg>
            </button>
            <div
              class="quantity"
            >
              <span
                class="actual-quantity-bad"
              >
                0
              </span>
              <span>
                 / 
              </span>
              <span
                class="expected-quantity"
              >
                1
              </span>
            </div>
            <button
              aria-label="remove item"
              class="change-quantity"
              disabled=""
              type="button"
            >
              <svg>
                minus.svg
              </svg>
            </button>
            <button
              aria-label="add item"
              class="change-quantity"
              type="button"
            >
              <svg>
                add.svg
              </svg>
            </button>
          </div>
        </div>
        <div
          class="item"
        >
          <div
            class="display-name"
          >
            Mefix roll (5cm x 10m)
          </div>
          <div
            class="controls"
          >
            <button
              aria-label="item information"
              class="item-info"
              type="button"
            >
              <svg>
                info-circle.svg
              </svg>
            </button>
            <div
              class="quantity"
            >
              <span
                class="actual-quantity-bad"
              >
                0
              </span>
              <span>
                 / 
              </span>
              <span
                class="expected-quantity"
              >
                1
              </span>
            </div>
            <button
              aria-label="remove item"
              class="change-quantity"
              disabled=""
              type="button"
            >
              <svg>
                minus.svg
              </svg>
            </button>
            <button
              aria-label="add item"
              class="change-quantity"
              type="button"
            >
              <svg>
                add.svg
              </svg>
            </button>
          </div>
        </div>
        <div
          class="item"
        >
          <div
            class="display-name"
          >
            Chest drain bottle
          </div>
          <div
            class="controls"
          >
            <button
              aria-label="item information"
              class="item-info"
              type="button"
            >
              <svg>
                info-circle.svg
              </svg>
            </button>
            <div
              class="quantity"
            >
              <span
                class="actual-quantity-bad"
              >
                0
              </span>
              <span>
                 / 
              </span>
              <span
                class="expected-quantity"
              >
                1
              </span>
            </div>
            <button
              aria-label="remove item"
              class="change-quantity"
              disabled=""
              type="button"
            >
              <svg>
                minus.svg
              </svg>
            </button>
            <button
              aria-label="add item"
              class="change-quantity"
              type="button"
            >
              <svg>
                add.svg
              </svg>
            </button>
          </div>
        </div>
        <div
          class="item"
        >
          <div
            class="display-name"
          >
            Chest drain tubing
          </div>
          <div
            class="controls"
          >
            <button
              aria-label="item information"
              class="item-info"
              type="button"
            >
              <svg>
                info-circle.svg
              </svg>
            </button>
            <div
              class="quantity"
            >
              <span
                class="actual-quantity-bad"
              >
                0
              </span>
              <span>
                 / 
              </span>
              <span
                class="expected-quantity"
              >
                1
              </span>
            </div>
            <button
              aria-label="remove item"
              class="change-quantity"
              disabled=""
              type="button"
            >
              <svg>
                minus.svg
              </svg>
            </button>
            <button
              aria-label="add item"
              class="change-quantity"
              type="button"
            >
              <svg>
                add.svg
              </svg>
            </button>
          </div>
        </div>
        <div
          class="item"
        >
          <div
            class="display-name"
          >
            Sterile water (H20) bottle (1000ml bottle)
          </div>
          <div
            class="controls"
          >
            <button
              aria-label="item information"
              class="item-info"
              type="button"
            >
              <svg>
                info-circle.svg
              </svg>
            </button>
            <div
              class="quantity"
            >
              <span
                class="actual-quantity-bad"
              >
                0
              </span>
              <span>
                 / 
              </span>
              <span
                class="expected-quantity"
              >
                1
              </span>
            </div>
            <button
              aria-label="remove item"
              class="change-quantity"
              disabled=""
              type="button"
            >
              <svg>
                minus.svg
              </svg>
            </button>
            <button
              aria-label="add item"
              class="change-quantity"
              type="button"
            >
              <svg>
                add.svg
              </svg>
            </button>
          </div>
        </div>
        <div
          class="item"
        >
          <div
            class="display-name"
          >
            Spencer wells forceps (Straight 20cm)
          </div>
          <div
            class="controls"
          >
            <button
              aria-label="item information"
              class="item-info"
              type="button"
            >
              <svg>
                info-circle.svg
              </svg>
            </button>
            <div
              class="quantity"
            >
              <span
                class="actual-quantity-bad"
              >
                0
              </span>
              <span>
                 / 
              </span>
              <span
                class="expected-quantity"
              >
                1
              </span>
            </div>
            <button
              aria-label="remove item"
              class="change-quantity"
              disabled=""
              type="button"
            >
              <svg>
                minus.svg
              </svg>
            </button>
            <button
              aria-label="add item"
              class="change-quantity"
              type="button"
            >
              <svg>
                add.svg
              </svg>
            </button>
          </div>
        </div>
      </div>
    </main>
    <footer>
      <button
        class="save"
        type="button"
      >
        Save
      </button>
    </footer>
  </div>
</div>
`;
